test/README.md

```diff
@@ -23,3 +23,103 @@ Some examples of test harnesses using these tests, from the source repository:
+its `Input` and `FD` values are determined as follows:
+
+1. Let `FD` be 0.
+1. Let `arg` be the resolved value of the _expression_ _operand_.
+1. If `arg` is the resolved value of an _expression_
+   with a `:test:function`, `:test:select`, or `:test:format` _annotation_
+   for which resolution has succeeded, then
+   1. Let `Input` be the `Input` value of `arg`.
+   1. Set `FD` to be `FD` value of `arg`.
+1. Else if `arg` is a numerical value
+   or a string matching the `number-literal` production, then
+   1. Let `Input` be the numerical value of `arg`.
+1. Else,
+   1. Emit "bad-input" _Resolution Error_.
+   1. Use a _fallback value_ as the resolved value of the _expression_.
+      Further steps of this algorithm are not followed.
+1. If the `fd` _option_ is set, then
+   1. If its value resolves to a numerical integer value 0 or 1
+      or their corresponding string representations `'0'` or `'1'`, then
+      1. Set `FD` to be the numerical value of the _option_.
+   1. Else if its value is not an unresolved value set by _option resolution_,
+      1. Emit "bad-option" _Resolution Error_.
+      1. Use a _fallback value_ as the resolved value of the _expression_.
+
+When `:test:function` is used as a _selector_,
+the behaviour of calling it as the `rv` value of MatchSelectorKeys(`rv`, `keys`)
+(see [Resolve Preferences](/spec/formatting.md#resolve-preferences) for more information)
+depends on its `Input` and `FD` values.
+
+- If the `Input` is 1 and `FD` is 1,
+  the method will return some slice of the list « `'1.0'`, `'1'` »,
+  depending on whether those values are included in `keys`.
+- If the `Input` is 1 and `FD` is 0,
+  the method will return the list « `'1'` » if `keys` includes `'1'`, or an empty list otherwise.
+- If the `Input` is any other value, the method will return an empty list.
+
+When an _expression_ with a `:test:function` _annotation_ is assigned to a _variable_ by a _declaration_
+and that _variable_ is used as an _option_ value,
+its resolved value is the `Input` value.
+
+When `:test:function` is used as a _formatter_,
+a _placeholder_ resolving to a value with a `:test:function` _expression_
+is formatted as a concatenation of the following parts:
+
+1. If `Input` is less than 0, the character `-` U+002D Hyphen-Minus.
+1. The truncated absolute integer value of `Input`, i.e. floor(abs(`Input`)),
+   formatted as a sequence of decimal digit characters (U+0030...U+0039).
+1. If `FD` is 1, then
+   1. The character `.` U+002E Full Stop.
+   1. The single decimal digit character representing the value floor((abs(`Input`) - floor(abs(`Input`))) \* 10)
+
+If the formatting target is a sequence of parts,
+each of the above parts will be emitted separately
+rather than being concatenated into a single string.
+
+
+### `:test:select`
+
+This _function_ accepts the same _operands_ and _options_,
+and behaves exactly the same as `:test:function`,
+except that it cannot be used for formatting.
```

Comment:
I think having a selection function that isn't capable of formatting is strange because formatting can be a precursor to selection, and that is indeed the case for plurals/numbers, and this test function takes numbers.

What is the motivation for having such a function for selection but not formatting?

One concern is that such a function implies that selection and formatting aren't linked. Another concern is that if the constraint that `:test:select` is defined by is not super useful, and if its functionality is a subset of the full `:test:function`, then we can remove `:test:select` altogether without loss of functionality.

Reply:
Ok, that makes sense. IIUC, the `:test:select` function isn't so much about number formatting as much as standing in for the category of functions that fit the bill of select-only, for which there exist regular examples like gender. Sounds good.